docs/src/operate/c8y/supported-operations.md

```diff
@@ -241,9 +243,177 @@ Arguments will be parsed correctly as long as following features are not include
 
 In case those unsupported shell features are present, the syntax that introduce them is interpreted literally.
 
-Be aware that SmartREST payload is always added as the last argument. The command presented above will actually lead to following code execution
+Be aware that SmartRest payload is always added as the last argument. The command presented above will actually lead to following code execution
 
 ```bash
 python /etc/tedge/operations/command.py $SMART_REST_PAYLOAD
 ```
-:::
\ No newline at end of file
+:::
+
+#### Create operation delivered via C8Y Json
+
+There are two approaches for handling custom operation via C8Y Json.
+The first one can only be used with main device, while the second one extends it to child devices. Here is a sample of incoming C8Y Json message with custom operation fragment on topic `c8y/devicecontrol/notifications`: 
```

Comment:
Before jumping into constraints and approach comparisons, it would be good to tell what are these two alternative approaches!

```suggestion
There are two approaches for handling custom operation via C8Y Json.
- the commands can be processed by a user-provided script
- the commands can be processed by the agent following a user-provided workflow
The first one can only be used with main device, while the second one extends it to child devices. Here is a sample of incoming C8Y Json message with custom operation fragment on topic `c8y/devicecontrol/notifications`: 
```

Reply:
Resolved by a267f04581ec93b1644543df4d45e998b590aaa8